Phaethon/ExternalApi/Controllers/InvoiceController.cs

```diff
@@ -5,48 +5,62 @@
 using System.Net;
 using System.Net.Http;
 using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Web;
 using System.Web.Http;
+using System.Web.Http.Cors;
 using Newtonsoft.Json;
 
 namespace ExternalApi.Controllers
 {
-    [RoutePrefix("Invoice")]
+    [EnableCors(origins: "http://localhost:49873", headers: "*", methods: "*")]
+    [RoutePrefix("Api/Invoice")]
     public class InvoiceController : ApiController
     {
-        [Route("Create")]
+        private readonly HttpClient _client;
+
+        public InvoiceController()
+        {
+            _client = new HttpClient();
+            _client.BaseAddress = new Uri("http://localhost:64007/Invoice/");
+        }
+
+        [Route("CreateOrUpdate")]
         [HttpPost]
-        public HttpResponseMessage Create([FromBody] Invoice invoice)
+        public async Task<HttpResponseMessage> CreateOrUpdate([FromBody] Invoice invoice)
         {
-            HttpClient client = new HttpClient();
-            var result = client.PostAsJsonAsync("http://localhost:64007/Invoice/Create", invoice).Result;
-            return result;
+            return await _client.PostAsJsonAsync("CreateOrUpdate", invoice);
         }
 
-        [Route("Read")]
+        [Route("GetInvoice")]
         [HttpGet]
-        public HttpResponseMessage Read(int id)
+        public async Task<HttpResponseMessage> GetInvoice(int id)
         {
-            HttpClient client = new HttpClient();
-            var result = client.GetAsync("http://localhost:64007/Invoice/Read?id=" + id).Result;
-            return result;
+            var parameters = HttpUtility.ParseQueryString(string.Empty);
+            parameters["id"] = id.ToString();
+            return await _client.GetAsync("GetInvoice?" + parameters);
         }
 
         [Route("GetInvoices")]
         [HttpGet]
-        public HttpResponseMessage GetInvoices()
+        public async Task<HttpResponseMessage> GetInvoices(int numOfRecords, int selectedCompany, string name, int selectedDate, string from, string to, string docNumber)
```

Comment:
I'd suggest you use a Filter object, instead of passing 7 parameters. (you can check my Job/ReadAll method, in the Job endpoint).
I think it would be much easier for you, when you want to change stuff.